pkg/gator/runner.go

```diff
@@ -86,7 +88,11 @@ func (r *Runner) skipTest(t Test) TestResult {
 func (r *Runner) runTest(ctx context.Context, suiteDir string, filter Filter, t Test) TestResult {
 	start := time.Now()
 
-	results, err := r.runCases(ctx, suiteDir, filter, t)
+	err := r.tryAddConstraint(ctx, suiteDir, t)
+	var results []CaseResult
+	if err == nil && !t.Invalid {
```

Comment:
It would be more clear to check for the invalid constraint error here and fail the test as-necessary here. That avoids double-checking t.Invalid and makes it easier to reason about what happens if the error is nil and t.Invalid is `true`.